Bottom: e20a47b26ea5ff74b80f2d0e2b1e1990bd4e7160
Top:    0563eb136ff1177a2b075a03f31e4bf86816bf2a
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2012-11-10 16:54:07 -0500

Temp fixes and diagnostics


---

diff --git a/src/polygon1.c b/src/polygon1.c
index d06030d..bcb9a52 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1120,7 +1120,11 @@ label_contour (PLINE * a)
       assert (label == INSIDE || label == OUTSIDE);
       LABEL_NODE (cur, label);
     }
-  while ((cur = cur->next) != first_labelled);
+  while ((cur = cur->next) != first_labelled &&
+	 !(cur->next == &a->head && first_labelled == NULL));
+
+  if (first_labelled == NULL)
+    fprintf (stderr, "OOPS, DIDN'T LABEL ANYTHING!\n");
 #ifdef DEBUG_ALL_LABELS
   print_labels (a);
   DEBUGP ("\n\n");
diff --git a/src/sweep.h b/src/sweep.h
new file mode 100644
index 0000000..e69de29
